src/Relationships/OneHasMany.php

```diff
@@ -15,31 +15,40 @@ class OneHasMany extends HasMany
 {
 	public function getEntitiesForPersistence()
 	{
-		$entities = [];
-		foreach ($this->toAdd as $add) {
-			$entities[] = $add;
-		}
-		foreach ($this->toRemove as $remove) {
-			if ($remove->isPersisted()) {
-				$entities[] = $remove;
-			}
-		}
 		if ($this->collection !== null || $this->wasLoaded) {
-			foreach ($this->getIterator() as $entity) {
-				$entities[] = $entity;
+			$entities = iterator_to_array($this->getIterator());
+
+		} else {
+			$entities = $this->added + $this->toAdd;
```

Comment:
Jednoduše, ty potřebuješ, aby `$author->books->add(new Book)` nepotřebovalo načíst do paměti úplně všechny knihy, co kdy ten daný autor napsal. Zároveň ale `$author->books` nikdy nesmí referenci na tu knihu ztratit (tj. když změníš název té knihy, tak `$model->persist($author)` musí tu knihu vždycky uložit, to do teď nefungovalo spolehlivě.